File: day4/C/max_chars.c
/**
  * @author : ashwek
  * @date : 25/12/2018
  */

#include<stdio.h>

void main(){

    char Str[50];
    int i, Max = 0;
    int possibleChar[94] = {0};  //ASCII 32-126

    printf("Enter a string = ");
    scanf("%[a-zA-Z-0-9 ]s", Str);  //Modify control string to accept blank spaces in input

    for(i=0; Str[i]!='\0'; i++){
        possibleChar[ (int)Str[i] - 32]++;
        if( possibleChar[Max] < possibleChar[ (int)Str[i] - 32] )
            Max = (int)Str[i] - 32;
    }

    printf("Most frequent character = \'%c\'\n", (Max+32));
}

File: day4/Python/MaxChars.py
"""
  @author : vishalshirke7
  @date : 25/12/2018
"""

input_str = input()
dictnry = {}
maxCount, maxChar = 0, ''
for char in input_str:
    if char in dictnry.keys():
        dictnry[char] += 1
    else:
        dictnry[char] = 1
    if dictnry[char] > maxCount:
        maxCount = dictnry[char]
        maxChar = char

print(maxChar)

